### DISCUSSION TITLE
Add coding style guidelines to Dev-Information

### DISCUSSION THREAD

**Discussion by gagarinlg** [EdgeTX/edgetx#2126] [General]
To ease the inclusion of PRs and reduce the style complaints when doing reviews we could do a few things:

- advertise the used coding style in the Dev-Wiki and all other relevant places (README.md?)
- add style configs for more indentation tools and editors, like Eclipse, VScode, ...?
- test and maybe implement a github action to check the coding style

Are there more ideas and opinions on that?

**Comment by elecpower** [comment]
dot point one "agree on the coding style"

**Reply by gagarinlg** [reply]
The style that is described by the clang config already in the repo?

**Reply by pfeerick** [reply]
It may be that the coding style for Companion needs to be set/documented, as a significant amount of the radio will be formatted per the `.clang-format` config file, which is automatically used by VSCode, and Eclipse and CLion can be configured to use it. 

It may be worth looking at `https://github.com/opentx/opentx/wiki/OpenTX-Code-Style-Guide` and using that as a starting point for own page. And as an more 'advanced' topic document how to do pre-commit formatting checks so it will done locally and you won't forget ;)

This is the action I have in mind atm... https://github.com/marketplace/actions/clang-format-github-action ... just need to see what the actual output is... I would hope it would basically error out if code isn't compliant, and point to what needs changing.

**Reply by elecpower** [reply]
Companion generally uses https://github.com/opentx/opentx/wiki/OpenTX-Code-Style-Guide though the work for yaml broke it as @raphaelcoeffic and my follow up bug fixes (for conformity) are .clang-format.
From what I can tell by checking some other style standards, OpenTX uses a hybrid style.
IMO all the project's code should use the same style and a standard one at that. Some PRs have both radio and Companion code and having different styles would be frustrating and likely problematic for the IDEs.
A page like the OpenTX one is needed so submitters can be referred before starting and then cannot complain if their PR is bounced for non-conformance whether that be automated or reviewers.
Having configs available for download for various common IDEs is a good idea too IMO

**Reply by gagarinlg** [reply]
I think style configs for the most common IDEs could be in the repo.
I don't like the idea of re-styling the whole companion code, though, because it creates a lot of noise or we would have to integrate all open PRs before. Also this makes moving code between OpenTX and EdgeTX harder. 
Gradually synchronising the styles of all software parts is probably easier to achieve.
